core/controllers/question_editor.py

```diff
@@ -115,22 +115,31 @@ class QuestionSkillLinkHandler(base.BaseHandler):
     """A handler for linking and unlinking questions to or from a skill."""
 
     @acl_decorators.can_manage_question_skill_status
-    def put(self, question_id, skill_id):
+    def put(self, question_id, skill_ids):
         """Updates the difficulty of the question with respect to the given
         skill.
         """
-        new_difficulty = float(self.payload.get('new_difficulty'))
-
-        question_services.update_question_skill_link_difficulty(
-            question_id, skill_id, new_difficulty)
+        skill_ids = skill_ids.split(',')
+
+        if self.payload.get('action') == 'update_difficulty':
+            new_difficulty = float(self.payload.get('new_difficulty'))
+            question_services.update_question_skill_link_difficulty(
+                question_id, skill_ids[0], new_difficulty)
```

Comment:
Actually this is also odd -- this is referencing one skill and the other is expecting multiple skills.

Seems like you need separate handlers here since you're expecting different data in each.

Reply:
Done